js/ui/drop_down_button.js

```diff
@@ -519,14 +523,18 @@ const DropDownButton = Widget.inherit({
                 break;
             case 'displayExpr':
                 this._compileDisplayGetter();
+                this._setListOption(name, value);
+                this._updateActionButton(this.option('selectedItem'));
                 break;
             case 'keyExpr':
                 this._compileKeyGetter();
+                this._setListOption(name, value);
                 break;
             case 'buttonGroupOptions':
                 this._innerWidgetOptionChanged(this._buttonGroup, args);
                 break;
             case 'dropDownOptions':
+                args.value.visible && delete args.value.visible;
```

Comment:
Is this change necessary? What test covers this?

Reply:
https://github.com/DevExpress/DevExtreme/pull/11847/files#diff-d9fa2b0f966e8a8b29e5b1d170e9ea05R599